### PR TITLE
test: ensure proper cleanup between config tests

### PR DESCRIPTION
**Pull request by watson**
This was all the code that came out of trying to solve a weird CI related issue in #1114.

I thought it was best to submit it as a separate PR in case we want to backport it.

### CODE REVIEW

File: lib/instrumentation/async-hooks.js [hunk 1]
```diff
@@ -3,10 +3,19 @@
...
+let _asyncHook
+
 module.exports = function (ins) {
   const asyncHook = asyncHooks.createHook({ init, before, destroy })
   const contexts = new WeakMap()
 
+  if (resettable) {
+    if (_asyncHook) _asyncHook.disable()
+    _asyncHook = asyncHook
+  }
```
Comment (Qard) [inline lib/instrumentation/async-hooks.js]: Is there a reason not to just _always_ do this? 🤔

Comment (watson) [reply lib/instrumentation/async-hooks.js]: Good question... I did so in the first iteration but I wasn't 100% sure if there were any implications of keeping a reference to this object so it couldn't be garbage collected. This way it only kept it around during testing - but it's probably irellevant 🤷‍♂

**Comment by watson** [comment]
jenkins run the tests please

File: lib/instrumentation/async-hooks.js (hunk 1 above)
Comment (Qard) [reply lib/instrumentation/async-hooks.js]: Fair. Approved anyway just to get the fix in. I can clean it up later. 🤔

**Comment by watson** [comment]
jenkins run the tests please

**Pull request merged**